internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,540 @@
+		ownerName:  prr.ownerName,
+		repoName:   prr.name,
+		repoID:     prr.rid,
+		branchName: newPR.Head,
+		id:         *pr.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) UpdatePullRequest(ctx context.Context, repo scm.Repository, pullReq scm.PullRequest, updatedPR scm.NewPullRequest) (scm.PullRequest, error) {
+	r := repo.(repository)
+	pr := pullReq.(pullRequest)
+
+	reviewerIDs, err := a.getUserIds(ctx, append(updatedPR.Reviewers, updatedPR.Assignees...))
+	if err != nil {
+		return nil, err
+	}
+
+	reviewers := make([]git.IdentityRefWithVote, len(reviewerIDs))
+	for i, reviewerID := range reviewerIDs {
+		rCopy := reviewerID
+		reviewers[i] = git.IdentityRefWithVote{
+			Id: &rCopy,
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	prTitle := updatedPR.Title
+	if updatedPR.Draft {
+		prTitle = "Draft: " + prTitle
+	}
+
+	activeLabel := true
+	labels := make([]core.WebApiTagDefinition, len(updatedPR.Labels))
+
+	for i, label := range updatedPR.Labels {
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+
+	srn := PrependPrefixIfNeeded(updatedPR.Head) // Pass the value of newPR.Head
+	trn := PrependPrefixIfNeeded(updatedPR.Base)
+
+	updateResult, err := a.gitClient.UpdatePullRequest(ctx, git.UpdatePullRequestArgs{
+		GitPullRequestToUpdate: &git.GitPullRequest{
+			Title:         &prTitle,
+			Description:   &updatedPR.Body,
+			SourceRefName: &srn,
+			TargetRefName: &trn,
+			Labels:        &labels,
+			Reviewers:     &reviewers,
+		},
+		RepositoryId:  &pr.repoID,
+		PullRequestId: &pr.id,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pullRequest{
+		ownerName:  r.ownerName,
+		repoName:   r.name,
+		repoID:     r.rid,
+		branchName: updatedPR.Head,
+		id:         *updateResult.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) getUserIds(ctx context.Context, usernames []string) ([]string, error) {
+	userIds := make([]string, len(usernames))
+
+	searchFilter := "General"
+
+	for i, username := range usernames {
+		uCopy := username
+		users, err := a.identityClient.ReadIdentities(ctx, identity.ReadIdentitiesArgs{
+			SearchFilter: &searchFilter,
+			FilterValue:  &uCopy,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+		if len(*users) != 1 {
+			return nil, fmt.Errorf("could not find user: %s", usernames[i])
+		}
+
+		userIds[i] = (*users)[0].Id.String()
+	}
+	return userIds, nil
+}
+
+func (a *AzureDevOps) GetPullRequests(ctx context.Context, branchName string) ([]scm.PullRequest, error) {
+	repos, err := a.getRepos(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	bn := PrependPrefixIfNeeded(branchName)
+
+	prs := []scm.PullRequest{}
+	for _, repo := range *repos {
+		pr, err := a.getPullRequest(ctx, bn, repo.Id.String())
+		if err != nil {
+			return nil, err
+		}
+		if pr == nil {
+			continue
+		}
+
+		prs = append(prs, convertPullRequest(pr))
+	}
+
+	return prs, nil
+}
+
+func convertPullRequest(pr *git.GitPullRequest) pullRequest {
```

Comment:
This should not be a pointer, since if it is. This method will panic

```suggestion
func convertPullRequest(pr git.GitPullRequest) pullRequest {
```